src/java.base/share/classes/java/nio/channels/FileChannel.java

```diff
@@ -959,7 +959,8 @@ public String toString() {
      *         If the preconditions on the parameters do not hold
      *
      * @throws UnsupportedOperationException
-     *         If an unsupported map mode is specified
+     *         If an unsupported map mode is specified, or if the implementation
```

Comment:
If you don't agree that the spec should be changed for the old API spec, let's remove that statement and only fix the new method's documentation.

Reply:
> If you don't agree that the spec should be changed for the old API spec, let's remove that statement and only fix the new method's documentation.

If we were to add the proposed `@implSpec`, then I don't think the `@throws UnsupportedOperationException` verbiage for `map(MapMode, long, long, MemorySession)` would need to be tweaked as the UOE is always thrown. If that would be acceptable, then perhaps the present issue could be revised and another separate issue created to address the optionality concerns that Alan raised, if that is something which it is desirable to investigate separately.

Reply:
I havo no strong preference where the information about this implemnation detail is written down.

I agree to change the original issue and add a new one about optionality.